dimod/constrained/constrained.py

```diff
@@ -688,6 +688,118 @@ def add_discrete_from_model(self,
         self.discrete.add(label)
         return label
 
+    def add_linear_constraints(self, A: ArrayLike, sense: Union[Sense, str], b: ArrayLike,
+                               *,
+                               variable_labels: Optional[Iterable[Variable]] = None,
+                               constraint_labels: Optional[Iterable[Hashable]] = None,
+                               ):
+        r"""Add multiple linear constraints to the model.
+
+        Add linear constraints defined by
+
+        .. math::
+
+                A_{ub}x \le b_{ub}, \\
+                A_{eq}x = b_{eq}
+
+        where `x` is a vector of decision variables,
+        :math:`b_{ub}` and :math:`b_{eq}` are vectors,
+        and :math:`A_{ub}` and :math:`A_{eq}` are matrices.
+
+        Args:
+            A: Each row of ``A`` specifies the coefficients of a linear constraint.
+            sense: One of `<=`, `>=`, `==`.
+            b: An array of left-hand-sides of the constraints.
+            variable_labels: The variable labels corresponding to the columns of `A`.
+                If ``None``, defaults to ``range(A.shape[1])``.
+            constraint_labels: The labels of the constraints. Labels of additional
+                constraints are generated.
+
+        Example:
+
+            Consider a binary linear program subject to the following constraints
+
+            .. math::
+                x_0 + 2x_1 - 7x_2 = 2 \\
+                x_0 + x_2 = 0 \\
+                -x_0 + x_1 \le 1 \\
+                3x_1 + x_2 \le 3
+
+            We can specify the constraints using matrices and vectors
+
+            >>> cqm = dimod.ConstrainedQuadraticModel()
+            >>> cqm.add_variables("BINARY", 3)
+            >>> A_eq = [[1, 2, -7], [1, 0, 1]]
+            >>> b_eq = [2, 0]
+            >>> cqm.add_linear_constraints(A_eq, "==", b_eq)
+            >>> A_ub = [[-1, 1, 0], [0, 3, 1]]
+            >>> b_ub = [1, 3]
+            >>> cqm.add_linear_constraints(A_ub, "<=", b_ub)
+            >>> print(cqm)
+            Constrained quadratic model: 3 variables, 4 constraints, 9 biases
+            <BLANKLINE>
+            Objective
+              0
+            <BLANKLINE>
+            Constraints
+              0: Binary(0) + 2*Binary(1) - 7*Binary(2) == 2.0
+              1: Binary(0) + Binary(2) == 0.0
+              2: -Binary(0) + Binary(1) <= 1.0
+              3: 3*Binary(1) + Binary(2) <= 3.0
+            <BLANKLINE>
+            Bounds
+            <BLANKLINE>
+
+            We can also use :class:`str`-labelled variables
```

Comment:
```suggestion
            You can also label the variables:
```

Is it helpful to point out that labels are strings? Would they be something else?

Reply:
I am distinguishing them from the int-labelled ones from before